### PR TITLE
pio resource cleanup

### PR DESCRIPTION
**Pull request by nmizukami**
Clean up PIO resources (decomposition etc). 

This prevent memory leak every time new netcdf is created.

### CODE REVIEW

**Review by ekluzek — APPROVED** [review]
This looks good to me. It's good practice to have subroutines to cleanup memory, so I endorse adding this.

I do suggest a bit about comments, but as things are pretty clear I don't think it's absolutely required.

**Comment by ekluzek** [comment]
pio_closefile is going to clean the file_desc_t.

Yes, we don't clean the iosystem_desc_t in CTSM, I think that might happen in the CMEPS mediator. But, it also should be OK to not clean iossystem_desc_t for the standalone case as it would only be done at the end of the simulation.

File: route/build/src/pio_utils.f90 [hunk 1]
```diff
@@ -15,7 +15,9 @@ MODULE pio_utils
   public::end_def
   public::inq_dim_len
   public::openFile
+  public::freeDecomp
```
[the remark this answers is not in the document]
Comment (nmizukami) [reply route/build/src/pio_utils.f90]: added the comments

**Comment by nmizukami** [comment]
I will do some small refactoring to keep iosystem_desc_t through the run (yes that make sense) in another pull request.

**Pull request merged**